### ISSUE TITLE
Feature Request: Support entities in _external_map

### ISSUE THREAD

**Issue by hemebond** [ericwa/ericw-tools#333]
The current `_external_map` implementation is great but it would be great to be able to combine more complex map files.

It would be great if it would just import an external map file in its entirety, not just the Worldspawn, and not do any conversions, merging entities, and brush entities, as-is.

**Comment by Paril** [comment]
The new `-add` feature can merge map files into the current one, but it can't be specified as an entity. I don't know if this feature would make much sense as an entity.. prefabs would be hard to do this way because of name conflicts

**Comment by hemebond** [comment]
> because of name conflicts

What kind of naming conflicts can you imagine?

**Comment by ericwa** [comment]
I guess if the external .map uses `target`/`targetname` for anything - e.g. if it contains a button targetting a door (all within the same external .map), that will break once several instances of the external .map are inserted into a main map - every button will target every door.

The only workaround for that is to use a better prefab system like TB's that lets you do selective edits to instances.

Despite that, this improvement to the `_external_map` system could still be useful for things that don't need target/targetname connections, but still need multiple entities (e.g. light fixtures, or prefabs that mix detail/structural), and it should be a quick change

**Comment by Paril** [comment]
We could add a "prefix/suffix" option to the _external_map entity, along with a list of keys that need to be prefixed. That would allow that to work, in theory.

**Comment by hemebond** [comment]
To be honest I wasn't even imagining something that complex; but rather just importing and transforming map files as-is.

I see more involved prefab support (prefixing entity properties, etc) to be the domain of the editor itself.

**Comment by Paril** [comment]
Yeah, I'm just thinking of ways people are going to use this and then complain when all of their doors open when they press a button, lol.

If it's just all simple entities though then yeah it should just work.

**Comment by pwitvoet** [comment]
For anyone that is interested in this feature, I made a .map processing tool named [MESS](https://pwitvoet.github.io/mess/index.html) _(Macro Entity Scripting System)_ that can insert the contents of other map files into a map, among other things. Similar to `misc_external_map`, it uses a special entity (`macro_insert`) to insert other maps, but it also supports local templates (`macro_template`) and it has entities for randomly placing multiple instances across a surface (`macro_cover`) or inside an area (`macro_fill`). It works by updating a .map file before it is being passed to the actual compile tools.

Entity name conflicts can be avoided by adding placeholders to entity names and targets in an external map, such as `door{targetname}`. The `{targetname}` placeholder is replaced by the `targetname` property of the macro entity that's creating the current instance. Placeholders can reference any property of the instance-creating entity - it's actually a scripting system that allows things like setting default values or automatically generating unique names.

The main use-cases I've found so far are for decorating terrain with props, and for abstracting complex entity setups into single 'template' entities.

**Comment by hemebond** [comment]
I don't think the compiler needs to worry about name conflicts in properties; at least for this enhancement. I think there are other solutions available for really advanced includes.

My requirement was just things like prefabs with lights, and other simple entities.
